docs/NodeDisruptionPolicy.md

```diff
@@ -0,0 +1,146 @@
+* Design for image-based updates (live apply and bootc, will be considered in the future)
+* Have the MCO validate whether a change can be successfully applied with the given NodeDisruptionPolicy (i.e. it is up to the responsibility of the user to ensure the correctness of their defined actions)
+
+## How it works right now
+The current cluster policies are always listed in NodeDisruptionPolicyStatus in a `MachineConfiguration`(not to be confused with `MachineConfig`) CR, called "cluster". All configurations should be applied to the `spec.nodeDisruptionPolicy` field of this CR. The operator will merge the user defined policies and the MCO defaults and then display them in the `status.nodeDisruptionPolicyStatus` field.
+
+Note: All configuration must be done to the CR named "cluster". Any `MachineConfiguration` object made with another name will be rejected via a ValidatingAdmissionPolicy. This will be the single point of control for all MachineConfig changes.
+
+For example, when there is no user defined policies, the status will just show the cluster defaults. See below:
+
+```console
+$ oc get MachineConfiguration/cluster -o yaml
+apiVersion: operator.openshift.io/v1
+kind: MachineConfiguration
+metadata:
+  creationTimestamp: "2024-04-16T15:02:37Z"
+  generation: 4
+  name: cluster
+  resourceVersion: "261205"
+  uid: 2c67b155-1898-452f-adbd-ed376afc0ea2
+spec:
+  logLevel: Normal
+  managementState: Managed
+  operatorLogLevel: Normal
+status:
+  nodeDisruptionPolicyStatus:
+    clusterPolicies:
+      files:
+      - actions:
+        - type: None
+        path: /etc/mco/internal-registry-pull-secret.json
+      - actions:
+        - type: None
+        path: /var/lib/kubelet/config.json
+      - actions:
+        - reload:
+            serviceName: crio.service
+          type: Reload
+        path: /etc/machine-config-daemon/no-reboot/containers-gpg.pub
+      - actions:
+        - reload:
+            serviceName: crio.service
+          type: Reload
+        path: /etc/containers/policy.json
+      - actions:
+        - type: Special
+        path: /etc/containers/registries.conf
+      sshkey:
+        actions:
+        - type: None
+  readyReplicas: 0
+```
+Say, for instance the user applied the following MachineConfiguration:
+```
+apiVersion: operator.openshift.io/v1
+kind: MachineConfiguration
+metadata:
+  name: cluster
+  namespace: openshift-machine-config-operator
+spec:
+  nodeDisruptionPolicy:
+    files:
+      - path: /etc/my-file
+        actions:
+          - type: None
+```
+Now, the Status will be updated to reflect the merged policy:
+```
+$ oc get MachineConfiguration/cluster -o yaml
+apiVersion: operator.openshift.io/v1
+kind: MachineConfiguration
+metadata:
+  creationTimestamp: "2024-04-16T15:02:37Z"
+  generation: 4
+  name: cluster
+  resourceVersion: "261205"
+  uid: 2c67b155-1898-452f-adbd-ed376afc0ea2
+spec:
+  logLevel: Normal
+  managementState: Managed
+  operatorLogLevel: Normal
```

Comment:
During local testing I vaguely recall that user applied nodeDisruption policy gets also displayed in spec field. If that is still the case, maybe we want to include it in the doc for correctness. If not, ignore this comment :)

Reply:
Good catch, totally missed that!